pkg/querier/querier_test.go

```diff
@@ -211,6 +416,130 @@ func Test_Series(t *testing.T) {
 	}, out.Msg.LabelsSet)
 }
 
+func Test_Series_WithLabelNameFiltering(t *testing.T) {
+	tests := []struct {
+		name                string
+		allowUtf8LabelNames bool
+		setCapabilities     bool
+		requestLabelNames   []string
+		expectedLabelNames  []string
+	}{
+		{
+			name:                "all label names pass through when UTF8 enabled",
+			allowUtf8LabelNames: true,
+			setCapabilities:     true,
+			requestLabelNames:   []string{"valid_name", "123invalid", "invalid-hyphen", "世界"},
+			expectedLabelNames:  []string{"valid_name", "123invalid", "invalid-hyphen", "世界"},
+		},
+		{
+			name:                "invalid label names filtered when UTF8 disabled",
+			allowUtf8LabelNames: false,
+			setCapabilities:     true,
+			requestLabelNames:   []string{"valid_name", "123invalid", "invalid-hyphen", "世界"},
+			expectedLabelNames:  []string{"valid_name"},
+		},
+		{
+			name:                "UTF8 labels filtered when UTF8 disabled",
+			allowUtf8LabelNames: false,
+			setCapabilities:     true,
+			requestLabelNames:   []string{"foo", "bar", "世界", "日本語"},
+			expectedLabelNames:  []string{"foo", "bar"},
+		},
+		{
+			name:               "filtering enabled when no capabilities set",
+			setCapabilities:    false,
+			requestLabelNames:  []string{"foo", "123invalid", "世界"},
+			expectedLabelNames: []string{"foo"},
+		},
+		{
+			name:                "all valid labels pass through",
+			allowUtf8LabelNames: false,
+			setCapabilities:     true,
+			requestLabelNames:   []string{"foo", "bar", "service_name"},
+			expectedLabelNames:  []string{"foo", "bar", "service_name"},
+		},
+		{
+			name:                "labels with dots pass through",
+			allowUtf8LabelNames: false,
+			setCapabilities:     true,
+			requestLabelNames:   []string{"service.name", "app.version"},
+			expectedLabelNames:  []string{"service.name", "app.version"},
+		},
+		{
+			name:                "empty label names with UTF8 disabled queries all labels and filters",
+			allowUtf8LabelNames: false,
+			setCapabilities:     true,
+			requestLabelNames:   []string{},
+			expectedLabelNames:  []string{"bar", "foo"},
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			foobarlabels := phlaremodel.NewLabelsBuilder(nil).Set("foo", "bar")
```

Comment:
Ideally this should contain utf8 label names

And the we should have a test case that tests it:
```
		{
			[...]
            allowUtf8LabelNames: true,
			requestLabelNames:   []string{},
			expectedLabelNames:  []string{"bar", "foo","日本語"},
		},
```